src/fw_update.c

```diff
@@ -243,32 +300,90 @@ static void fw_update_thread(void *arg)
                                             GOLIOTH_SYS_WAIT_FOREVER);
 
         uint64_t start_time_ms = golioth_sys_now_ms();
-        size_t bytes_downloaded = 0;
+        download_ctx.bytes_downloaded = 0;
+        uint32_t next_block = 0;
+
+        if (!download_ctx.sha)
+        {
+            golioth_sys_sha256_destroy(download_ctx.sha);
+        }
+        download_ctx.sha = golioth_sys_sha256_create();
```

Comment:
Are we destroying this sha anywhere other than above? We should destroy it anytime we complete or fail a firmware update.